packages/seacas/libraries/ioss/src/exodus/Ioex_ChangeSet.C

```diff
@@ -0,0 +1,160 @@
+// Copyright(C) 1999-2020, 2022, 2023 National Technology & Engineering Solutions
+// of Sandia, LLC (NTESS).  Under the terms of Contract DE-NA0003525 with
+// NTESS, the U.S. Government retains certain rights in this software.
+//
+// See packages/seacas/LICENSE for details
+
+#include "Ioex_ChangeSet.h"
+#include "Ioex_DatabaseIO.h"
+
+#include "Ioss_CodeTypes.h" // for Int64Vector, IntVector
+#include "Ioss_SmartAssert.h"
+#include <cassert> // for assert
+#include <cmath>   // for sqrt
+#include <fmt/format.h>
+#include <fmt/ostream.h>
+#include <iostream> // for ostringstream
+#include <stdlib.h>
+#include <string> // for string, operator==, etc
+#include <vector>
+
+#include "Ioss_ChangeSetFactory.h"// for ChangeSetFactory
+#include "Ioss_Property.h"        // for Property
+#include "Ioss_PropertyManager.h" // for PropertyManager
+#include "Ioss_Region.h"          // for Region
+
+namespace Ioex {
+  // ========================================================================
+  const ChangeSetFactory *ChangeSetFactory::factory()
+  {
+    static ChangeSetFactory registerThis;
+    return &registerThis;
+  }
+
+  ChangeSetFactory::ChangeSetFactory() : Ioss::ChangeSetFactory("exodus")
+  {
+    Ioss::ChangeSetFactory::alias("exodus", "exodusii");
+    Ioss::ChangeSetFactory::alias("exodus", "exodusII");
+    Ioss::ChangeSetFactory::alias("exodus", "genesis");
+#if defined(PARALLEL_AWARE_EXODUS)
+    Ioss::ChangeSetFactory::alias("exodus", "dof_exodus");
+    Ioss::ChangeSetFactory::alias("exodus", "dof");
+#endif
+  }
+
+  Ioss::ChangeSet *ChangeSetFactory::make_ChangeSet(Ioss::Region* region) const
+  {
+    return new ChangeSet(region);
+  }
+
+  Ioss::ChangeSet *ChangeSetFactory::make_ChangeSet(Ioss::DatabaseIO* db,
+                                                    const std::string& dbName,
+                                                    const std::string& dbType,
+                                                    unsigned fileCyclicCount) const
+  {
+    return new ChangeSet(db, dbName, dbType, fileCyclicCount);
+  }
+
+  // ========================================================================
+  ChangeSet::ChangeSet(Ioss::Region *region)
+      : Ioss::ChangeSet(region)
+  {
+    Ioss::ChangeSet::m_supportedFormats = (Ioss::CHANGE_SET_INTERNAL_FILES     |
```

Comment:
Should this be set just once as a member initializer in the .h file...

Reply:
That is an inherited class member ... how would I set it in the .h file ? I can see setting it as part of the constructor initialization ...